File: Codeforces/Golang/800-1000/339A - Helpful Maths/main.go
package main

import (
	"fmt"
	"slices"
	"strings"
)

func main() {
	var s string
	fmt.Scan(&s)
	parts := strings.Split(s, "+")
	slices.Sort(parts)
	if len(parts) == 1 {
		fmt.Println(parts[0])
	} else {
		fmt.Println(strings.Join(parts, "+"))
	}

}

File: Codeforces/Rust/800-1000/339A - Helpful Maths/src/main.rs
use std::io;

fn main() {
    let mut input = String::new();
    let _ = io::stdin().read_line(&mut input);

    let s = input.trim();
    let mut parts: Vec<&str> = s.split('+').collect();
    parts.sort();
    let result = parts.join("+");
    if result.len() == 1 {
        println!("{}", result)
    } else {
        println!("{}", result)
    }
}
